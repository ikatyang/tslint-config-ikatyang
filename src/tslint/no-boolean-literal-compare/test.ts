declare const boolean: boolean;
declare const booleanOrUndefined: boolean | undefined;

// good

if (boolean) {

}

if (booleanOrUndefined === false) {

}

// bad

if (boolean === true) {

}
