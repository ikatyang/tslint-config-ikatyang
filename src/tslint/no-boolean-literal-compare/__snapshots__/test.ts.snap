// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
declare const boolean: boolean;
declare const booleanOrUndefined: boolean | undefined;

// good

if (boolean) {

}

if (booleanOrUndefined === false) {

}

// bad

if (boolean === true) {
    ~~~~~~~             [This expression is unnecessarily compared to a boolean. Just use it directly.]

}

`;
